PowerEditor/src/tools/NppShell/src/NppShell.cpp

```diff
@@ -200,22 +212,26 @@ BOOL RegisterServer() {
 //---------------------------------------------------------------------------
 // UnregisterServer
 //---------------------------------------------------------------------------
-BOOL UnregisterServer() {
+BOOL UnregisterServer(BOOL bUserMode) {
 	TCHAR szKeyTemp[MAX_PATH + GUID_STRING_SIZE];
 
-	RegDeleteKey(HKEY_CLASSES_ROOT, szShellExtensionKey);
+	HKEY hRootKey = (bUserMode) ? HKEY_CURRENT_USER : HKEY_CLASSES_ROOT;
+	wchar_t* szPrefix = (bUserMode) ? L"SOFTWARE\\Classes\\" : L"";

```

Comment:
Should use HKEY_LOCAL_MACHINE instead of HKEY_CLASSES_ROOT. This will remove requirement of making `szPrefix` dependant of `bUsermode` but always use the _SOFTWARE\Classes_ subkey.

Reply:
No, Users who don't have administrative privileges cannot modify the HKLM.

Reply:
Sorry if I might sound rude, but users also don't have "administrative privileges" in HKCR. My point is, that writing to HKCR in it's entirety is a bad idea, since it's an aggrregated/composite node [(see here)](https://docs.microsoft.com/en-us/windows/win32/sysinfo/hkey-classes-root-key?redirectedfrom=MSDN).

The linked article states everything that is needed to endorse my comment, so I won't bother elaborating my request any further.